src/System.Linq/tests/MinMaxTests.cs

```diff
@@ -0,0 +1,373 @@
+﻿using System.Collections.Generic;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public class MinMaxTests
+    {
+        [Fact]
+        public void MinInt()
+        {
+            var one = Enumerable.Range(1, 10).ToArray();
+            var minusTen = new[] { -1, -10, 10, 200, 1000 };
+            var hundred = new[] { 3000, 100, 200, 1000 };
+            Assert.Equal(1, one.Min());
+            Assert.Equal(-10, minusTen.Min());
+            Assert.Equal(100, hundred.Min());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<int>().Min());
+        }
+
+        [Fact]
+        public void MaxInt()
+        {
+            var ten = Enumerable.Range(1, 10).ToArray();
+            var minusTen = new[] { -100, -15, -50, -10 };
+            var thousand = new[] { -16, 0, 50, 100, 1000 };
+            Assert.Equal(10, ten.Max());
+            Assert.Equal(-10, minusTen.Max());
+            Assert.Equal(1000, thousand.Max());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<int>().Max());
+        }
+
+        [Fact]
+        public void MinLong()
+        {
+            var one = Enumerable.Range(1, 10).Select(i => (long)i).ToArray();
+            var minusTen = new[] { -1L, -10, 10, 200, 1000 };
+            var hundred = new[] { 3000L, 100, 200, 1000 };
+            Assert.Equal(1, one.Min());
+            Assert.Equal(-10, minusTen.Min());
+            Assert.Equal(100, hundred.Min());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<long>().Min());
+        }
+
+        [Fact]
+        public void MaxLong()
+        {
+            var ten = Enumerable.Range(1, 10).Select(i => (long)i).ToArray();
+            var minusTen = new[] { -100L, -15, -50, -10 };
+            var thousand = new[] { -16L, 0, 50, 100, 1000 };
+            Assert.Equal(10, ten.Max());
+            Assert.Equal(-10, minusTen.Max());
+            Assert.Equal(1000, thousand.Max());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<long>().Max());
+        }
+
+        [Fact]
+        public void MinSingle()
```

Comment:
The names of these functions are mixing the C# keywords with the type names, e.g. int and long vs Single.  Ideally they'd be consistent with one or the other, e.g. Int32, Int64, Single vs Int, Long, Float